engine/src/cubos/engine/tools/palette_editor/plugin.cpp

```diff
@@ -0,0 +1,79 @@
+#include <cubos/core/data/old/debug_serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/renderer/plugin.hpp>
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+#include <cubos/engine/tools/palette_editor/plugin.hpp>
+#include <cubos/engine/voxels/palette.hpp>
+#include <cubos/engine/voxels/plugin.hpp>
+
+#include "imgui.h"
+
+using cubos::core::data::old::Debug;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using namespace cubos::engine;
+
+struct SelectedPaletteInfo
+{
+    VoxelPalette palette;
+};
+
+static void checkAssetEventSystem(cubos::core::ecs::EventReader<tools::AssetSelectedEvent> reader, Read<Assets> assets,
+                                  Write<SelectedPaletteInfo> selectedPalette)
+{
+    for (const auto& event : reader)
+    {
+        if (assets->type(event.asset) == typeid(VoxelPalette))
+        {
+            CUBOS_INFO("Opening Palette Editor for {} ...", Debug(event.asset));
+            const Asset<VoxelPalette> paletteAsset = AnyAsset(event.asset);
+            selectedPalette->palette = assets->read(paletteAsset).get();
+        }
+    }
+}
+
+static void paletteEditor(Write<Renderer> renderer, Write<SelectedPaletteInfo> selectedPalette)
+{
+    auto& palette = selectedPalette->palette;
+
+    if (palette.data() == nullptr)
+    {
+        return; // no data on the palette yet!
+    }
+
+    ImGui::Begin("Palette Editor");
+
+    for (uint16_t materialIndex = 1; materialIndex < palette.size(); ++materialIndex)
+    {
+        const VoxelMaterial& material = palette.get(materialIndex);
+
+        std::string label = "Material " + std::to_string(materialIndex);
+        if (ImGui::ColorEdit4(label.c_str(), (float*)&material.color.r))
+        {
+            VoxelMaterial newMaterial = material;
+            palette.set(materialIndex, newMaterial);
+        }
+    }
+
+    if (ImGui::Button("Set this Palette as default"))
+    {
+        CUBOS_INFO("New palette set: {}", Debug(palette));
+        (*renderer)->setPalette(palette);
+    }
+
+    ImGui::End();
+}
+
+void cubos::engine::tools::paletteEditorPlugin(Cubos& cubos)
+{
+    cubos.addPlugin(imguiPlugin);
+    cubos.addPlugin(voxelsPlugin);
```

Comment:
```suggestion
    cubos.addPlugin(rendererPlugin);
```

You depend on the `rendererPlugin`, which in turn depends on `voxelsPlugin`.